### PR TITLE
Add Melding of the Flesh

### PR DESCRIPTION
**Pull request by learn2draw**
Adds support for the new Siege Unique `Melding of the Flesh`

### Link to a build that showcases this PR:
https://pastebin.com/6mdPAVwE

### Before screenshot:
image

### After screenshot:
image

### CODE REVIEW

**Comment by learn2draw** [comment]
image


Loreweave still sets maximum resistances properly

**Comment by zao** [comment]
Do we display maximum resistance anywhere else in the UI and is there a need to show what the cap is now that it may differ from the maximum resistance?
I'm thinking of places like Calcs breakdowns and tooltips.

**Comment by learn2draw** [comment]
> Do we display maximum resistance anywhere else in the UI and is there a need to show what the cap is now that it may differ from the maximum resistance? I'm thinking of places like Calcs breakdowns and tooltips.

Slight change made to now use a new mod override with `Melding of the Flesh` being a source for Maximum resist (and showing which element was the highest, and now being used for the cap)
image

**Comment by learn2draw** [comment]
https://www.reddit.com/r/pathofexile/comments/s78e4x/-/hta1h9m/?context=3

File: src/Modules/CalcDefence.lua [hunk 1]
```diff
@@ -81,10 +81,29 @@ function calcs.defence(env, actor)
 	output.DamageReductionMax = modDB:Override(nil, "DamageReductionMax") or data.misc.DamageReductionCap
 	output.PhysicalResist = m_min(output.DamageReductionMax, modDB:Sum("BASE", nil, "PhysicalDamageReduction"))
 	output.PhysicalResistWhenHit = m_min(output.DamageReductionMax, output.PhysicalResist + modDB:Sum("BASE", nil, "PhysicalDamageReductionWhenHit"))
+
+	-- Highest Maximum Elemental Resistance for Melding of the Flesh
+	local highestResistMax = 0;
+	local highestResistMaxType = "";
```
Comment (Wires77) [inline src/Modules/CalcDefence.lua:87]: These variables can be tucked inside the if statement since they're not needed outside of it

**Review by Wires77 — APPROVED** [review]
Made a couple other nitpicky fixes, but otherwise LGTM

**Pull request merged**